src/resources/views/fields/latlng.blade.php

```diff
@@ -0,0 +1,204 @@
+        <label>{{ $field['label'] }}</label>
+        <div class="input-group">          
+            <div class="input-group-addon">
+                <i class="fa fa-map-marker"></i>
+            </div>
+            <input id="search_address" class="form-control" type="text" placeholder="Enter a location" />
+            <div id="geolocate" class="input-group-addon btn btn-primary">
+                <i class="fa {{$field['geolocate_icon']}}"></i>
+            </div>
+        </div>
+    </div>
+    <div class="form-group col-md-4">
+        <label>&nbsp;</label>
+        <input
+        type="text"
+        id="latlong"
+        class="form-control"
+        @foreach ($field as $attribute => $value)
+            @if (is_string($attribute) && is_string($value))
+               @if($attribute == 'value')
+                    {{ $attribute }}="{{ old($field['name']) ? old($field['name']) : $value }}"
+                @else
+                    {{ $attribute }}="{{ $value }}"
+                @endif
+            @endif
+        @endforeach
+        >
+    </div>
+    <div class="form-group col-md-12">
+        <div id="map-canvas" style="{{$field['map_style']}}"></div>  
+    </div>
+
+{{-- ########################################## --}}
+{{-- Extra CSS and JS for this particular field --}}
+{{-- If a field type is shown multiple times on a form, the CSS and JS will only be loaded once --}}
+@if ($crud->checkIfFieldIsFirstOfItsType($field, $fields))
+
+    {{-- FIELD CSS - will be loaded in the after_styles section --}}
+    @push('crud_fields_styles')
+	    {{-- YOUR CSS HERE --}}
+    @endpush
+
+    {{-- FIELD JS - will be loaded in the after_scripts section --}}
+    @push('crud_fields_scripts')
+    	{{-- YOUR JS HERE --}}
+        <script src="https://maps.googleapis.com/maps/api/js?v=3&libraries=places&key={{$field['google_api_key']}}"></script>
+
+        <script>
+        $(document).ready(function() {
+
+            // overwrite BackPack-CRUD submit on Enter
+            $(window).keydown(function(event){
+                if(event.keyCode == 13) {
+                    event.preventDefault();
+                    return false;
+                }
+            });
+
+            function initialize() {
+                var marker;
+                var default_zoom = {{$field['default_zoom']}};
+                var geolocate_icon = "{{$field['geolocate_icon']}}";
+                var latlong = document.getElementById('latlong').value; // get latlng value if any
+                latlong = latlong.split(',');
+                var latlng = (latlong == '') ? new google.maps.LatLng(3.138675,101.6167769) : new google.maps.LatLng(latlong[0], latlong[1]);
+                var geocoder = new google.maps.Geocoder();
+                var mapOptions = {
+                    center: latlng,
+                    zoom: 8,
+                    mapTypeId: google.maps.MapTypeId.ROADMAP,
+                    draggableCursor: "pointer",
```

Comment:
Some might want to customise the options?

Reply:
Good idea. I'll look to it later. But as a user, I think some setting is better set with default value. Otherwise to much information in crud->addField().